### ISSUE TITLE
Building on indestructible grids

### ISSUE THREAD

**Issue by devu** [KeenSoftwareHouse/SpaceEngineers#292]
It should not be allowed to build on grids you cannot destroy. Unless by admin of the server. In fact Indestructible nature of grid could be omitted if done by admin as well.

**Comment by mexmer** [comment]
if you have complaint or suggestion about new feature ... use forums ... github is not for discussing user suggestions.

**Comment by JimLess** [comment]
agree with devu, this is not suggestion, this is bug. u can grind indestructible blocks. u can weld it, u can build on it. it's bug...

**Comment by devu** [comment]
Well, It worked that way only last Thursday.
Instead addressing it, they turned the grinder on for indestructible grids.
I really hope this will come back. One of most 'must-have' features for
public servers with some more game play.

**Comment by mexmer** [comment]
it's not community made change.
if you implore it's bug, then report it on forums, that's one and only official way to report bugs.

**Comment by devu** [comment]
Jump drives are not community made changes and you are still interested about it. I thought that would be something that community could pick up too.

**Comment by JimLess** [comment]
i still wait this bugfix. so some must do this work...

**Comment by Spartan322** [comment]
This isn't a github issue, this does not belong here, go to the forums.

**Comment by devu** [comment]
But community could look into it. It's more asking for help rather
reporting. Keen wasn't able to look at this just yet ad it was reported
many times already.
If I understand correctly and everyone can see it, community made changes
sometimes getting adopted.
I do not understand why you were so quick to send me out with this
particular bit. While the other bits are being worked on.

Again, did Jump Drive cluster issue is a community made bug? No, but you
picked this up. I simply don't get it.
I can't go and report it to Keen because it was already done. Then I will
get banished for not reading already reported bugs :D

Maybe I started wrong. Let me try again.

Could community look into this problem please?

**Comment by Spartan322** [comment]
Not every bug is to be reported here, most (if not all) go to the forums, also, you didn't even start off stating it is a bug, you stated a suggestion, this is not a suggestion box, please go to the forums.  Only code related suggestions go here (of which would be appropriate in the forums as well)

**Comment by devu** [comment]
I see plenty of reports and suggestions here too. But ok, probably I am not
a part of a circle of trust here. Thanks for clarification.
This subject will stuck in the limbo then, because it was already reported
and nobody is willing to pick this up.

**Comment by zrisher** [comment]
This is definitely a bug, but I think I see where they're drawing the line, maybe this is accurate and can help people understand what goes where:
* Any PR goes on Github (obviously)
* Issues can be opened on Github IFF they directly reference the code and discuss it, or are directly related to an open PR. Otherwise they must go on the forums.

Of course, I've still never seen anything official stating we can't use github to report bugs and feature requests. But it's easier not to argue.

**Comment by Spartan322** [comment]
its more a precedent, it is a commonality on github (and mot SVN sites), and a lot of profession SVN users get annoyed when you don't follow the above rule (unless a readme or something of the like says otherwise)

**Comment by zrisher** [comment]
Every open source project I've worked on uses Github issues for all issue tracking, relying on an external system seems a little strange to me especially when that system is a forum. But like I said, happy to work with it, not going to argue.

**Comment by mexmer** [comment]
OK seems i need to emphasize it again.

ONLY OFFICIAL CHANNEL TO REPORTING BUGS AND POSTING SUGGESTION FOR GAME IS FORUM.

Therefore do not discuss, or open issues here, that are not related to code changes made by community or that do not directly relate to code changes you will make/plan to make.

**Comment by FavioGalvis** [comment]
mexmer, are you a keen developper? if not why are you so entitled to say that?

**Comment by Spartan322** [comment]
@FavioGalvis this is not the place to discuss this kind of stuff, this should be closed because this is not to place to be discussing this

**Comment by devu** [comment]
You have to understand that for average Jo your procedures seems to be
unclear. Link to the source code is official and public, so is the list of
issues in there.
You reading them and getting impression that's what Exactly you should do
too. Don't you agree?
I see plenty of bug reports, suggestions that nobody care is caused by who
but the fact they exist in game.

Go for it. Remove it. If that became a issue on its own for you guys just
remove unwanted threads or lock possibility to do so if you getting
frustrated about it.
Sorry for creating such debate. It is simply natural to use features built
into the system the way the were designed for.

If you have to clarify them each time such a way and waste your time, it
might be worth to state it somewhere.
I'll stay away from this place then.

**Comment by FavioGalvis** [comment]
@devu they cant remove it, they are not part of KSH, therefore they dont put rules or deny users from their permision to write a report.

Making this clear:
A. Keen Software house HAS NOT stablished any workflow for the bug reports
B. http://forum.keenswh.com/threads/bug-reports-info.6584182/ Post its a suggestion for a place to report bugs
C. Github users as you and me can't close issues nor delete them (only the user making the report).

And yes this is a bug in the code of this Proyect where the user (and i quote you @devu) "should not be allowed to build on grids you cannot destroy. Unless by admin of the server. In fact Indestructible nature of grid could be omitted if done by admin as well."

**Comment by Spartan322** [comment]
@FavioGalvis How is that quote not a suggestion? Also this does not ask to improve the code or explain a bug in the code, nor does it point out an issue and suggest soultions, most of the bugs and suggestion here are code releated only, that is not the purpose of github's issue tracker, we have a [bug](http://forums.keenswh.com/forums/bug-reports.326950/) and [suggestion](http://forums.keenswh.com/forums/suggestions.313735/) section for a reason

**Comment by FavioGalvis** [comment]
@Spartan322 again you are not contributing, entitled making demands that this should not be reported here, and not helping, this place, as you call it, its not exclusive for people that "can" read code, you don't even know if the rest of the people do or not, ITS RUDE TO ASUME THE DONT KNOW AND TRY TO DISCRIMINATE THEM.

**Comment by Spartan322** [comment]
There is a purpose to posting most of a game's source to github, cause it isn't for those who just want rails, and it isn't for people who don't know how to read C#, there is no purpose for people who can't read to be here, if there is a reason, tell me, cause it isn't for random suggestion and bugs that are meant for the forums.

**Comment by devu** [comment]
It is purely about logic behind this feature and is more a functional bug.
There is nothing wrong with the part of existing code that controls it. It
just needs further improvement.

How I would do it from code point of view?
Each time build event getting fired check against grid indestructible flag
and prevent/cancel it when that flag is set to true.

Otherwise not even admin but anybody can place any block by mistake (or
not) and nobody will be able to remove it from the grid.

**Comment by FavioGalvis** [comment]
why you make all that assumtions? jesus, you talk like you are in the head of everyone around here.

Just put it simple, if you dont like this Report made from @devu, then dont make any coments here. There is no need to show you are discriminating users, just stay off the treads you dont think are productive for you. You have no autority to say this is not a place for X person, Without knowing if he/she knows C#. Thats discrimitantion.

**Comment by Spartan322** [comment]
Stop hijacking.

**Comment by sunkin351** [comment]
May I add, that if this is a suggestion as you say it is, that it should be in the suggestions section?

**Comment by FavioGalvis** [comment]
@sunkin351 the post on the forums 'is' the suggestion. not this tread.

And this report does not look like a suggestion, if a player is able to interact (construction/deconstruction) with protected grids with out the admin permision, and without hackind the conexion, then the protection on the grid itself is not working.

**Comment by Spartan322** [comment]
Stop Hijacking the issue

**Comment by FavioGalvis** [comment]
@Spartan322 well you can say "this should not be here".

I say it should, for the points adressed above.

**Comment by devu** [comment]
To clarify I can read and understand C#. I wish to have more time on my hands to look into that myself. Looks like I will have no other choice. 

After reading all this thread again it is the ONLY one that met such oppose from community members.

I think some here accusing others to be stupid but fails to understand the merit of the given problem if the very issue shows luck of logical thinking should consider to give up programming in any programming language.

**Comment by Spartan322** [comment]
I'm gonna go read the dragon book now.

**Comment by FavioGalvis** [comment]
@devu :+1:
Aye.

**Comment by mexmer** [comment]
@FavioGalvis no, i'm not dev, but i've been in contact with them.

As for bugs, every bug is tested and then reposted to their internal tracking system by phand (you could see her responding on forums bug section qute often), and one other person in charge of forums. but phand and that other person are only picking up reports from forums.

same goes for suggestion.

**Comment by devu** [comment]
Well, the way Keen solved it next day after reporting it was to add ability to grind down indestructible grid. Easy and quick solution to the problem. SO they have seen that report for sure. Such fix doesn't solve anything. Because from logical point of view being able to grind down indestructible grid doesn't make any sense. My goal will be to make it a proper way. But I will try this on my own if time permits. 

I am running public server and as Admin begging for this feature to make my server playable. Such feature can be done very inefficient way using modding API. I tried, but gave up on changing code every Thursday. To see how it performs is also not encouraging me to go down this road. I would prefer to see such feature to be build in.

I wish Keen would stay focused on improving possibilities of customizing DS instead creating artificial game mode with scenarios that nobody seems to be using. But those features dedicated for it, would be much more appreciated by online server community that is a 2nd driving force after mods to popularize sandbox type game.

So I saw this very feature as one of the most awaited by admins and those who run public servers. I even closed that proposal on their forum thanking for this the same day feature arrived. 

...but it wasn't complete, in fact it was reversed. That's why I thought community could step in and help to make something very useful here. Because the way Keen tackled the problem didn't solve anything. 

Instead I was sent out back, now having ridiculous conversation on Friday evening. Personally I know so many people in admin community giving up their plans on running their custom created SE worlds simply because they were waiting for a basic things from more than a year.  Less and less people believe there is anything to wait for. 

Single player with scenarios is not what majority of people bough this game for.

**Comment by devu** [comment]
And to add tho this if you prefer speaking code related stuff here. 

Today's patch by Tyrsis
MyAPIGateway.Session.DamageSystem interface could help to improve this situation on modding API side but I'm very concerned about performance if extra overhead was introduced to the servers that already suffer. This 'suggestion' of mine is still valid for performance reasons.

**Comment by devu** [comment]
And as much as I'm grateful for Tyrsis contribution his addition today doubled average transfer on DS firing too many events no matter you use it or not. This is why I see importance of making such things on low level. To be honest to see quality of code here I will soon become one of those who gave up.

**Comment by zrisher** [comment]
@devu I totally understand your frustrations, I think a lot of us are in the same boat. I also see dedicated servers as the main way forward for growing the user base of this game. And they are very under-served by current official development efforts.

However, the fact that core is open source and accepting PRs means that as a community we **can** get the things done we need done, given enough time and effort. I know Keen is behind on incorporating some important fixes and changes provided by PRs here, but I believe they probably recognize the power and importance of our ability to make changes, and they will eventually get through their backlog. Otherwise they never would have gone through the effort and risks of opening this up.

TLDR: we can do it, together. But I totally understand that the amount of work resting on our shoulders, in the short term anyway, turns people off to the game and developing for it. I try to think of the developing itself as a game, and if I'm not enjoying myself there's no point doing it.

And I know we're venturing into discussion territory here and probably going to get yelled at, happy to speak with any of you over steam, I'm [uRxP]DrChocolate.

**Comment by sunkin351** [comment]
What did you say about this last update? Is that what's causing the issue I have been having ever since my game updated?

**Comment by sunkin351** [comment]
It kinda goes like this: I start up my game, I enter my save game and start up, and as I move, there is a predictable interval where my movement would come to nearly a complete stop for about a fraction of a second, and the interval time is about once every 2 seconds. I noticed this "lag" also effects every other game mechanic in terms of timing, most noticeably when mining and in the assembler. I am wondering if its just me.
